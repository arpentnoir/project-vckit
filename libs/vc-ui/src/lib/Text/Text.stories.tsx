import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Text } from './Text';

const Story: ComponentMeta<typeof Text> = {
  component: Text,
  title: 'Text',
};
export default Story;

export const Default: ComponentStory<typeof Text> = () => <Text>Default</Text>;

export const Header5: ComponentStory<typeof Text> = () => (
  <Text variant="h5">Header5</Text>
);
