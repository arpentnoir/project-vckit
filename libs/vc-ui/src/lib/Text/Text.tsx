import { Typography, TypographyProps } from '@mui/material';
import React from 'react';

export const Text = React.forwardRef<HTMLElement, TypographyProps>(
  ({ children, variant = 'body2', ...rest }, ref) => {
    return (
      <Typography variant={variant} tabIndex={0} {...rest} ref={ref}>
        {children}
      </Typography>
    );
  }
);
