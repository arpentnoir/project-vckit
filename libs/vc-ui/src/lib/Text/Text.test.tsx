import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Text } from './Text';

describe('Text', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Text>Soopa Doopa</Text>);

    expect(baseElement).toMatchSnapshot();
  });
});
