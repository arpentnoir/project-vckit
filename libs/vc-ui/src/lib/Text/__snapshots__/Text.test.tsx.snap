// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
<body>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      tabindex="0"
    >
      Soopa Doopa
    </p>
  </div>
</body>
`;
